src/test/user-query.test.ts

```diff
@@ -0,0 +1,75 @@
+import { expect } from "chai";
+import request, { SuperTest, Test } from "supertest";
+import { getRepository, Repository } from "typeorm";
+
+import { User } from "../entity";
+import { ErrorMessage } from "../core/error";
+import { JWTService } from "../core/security/jwt";
+import { CreateUserInput } from "../schema/schema.types";
+import { createUserEntity, requestQuery } from "./common";
+
+describe("Tests for User", () => {
+  let agent: SuperTest<Test>;
+  let repository: Repository<User>;
+
+  const user: CreateUserInput = {
+    name: "user",
+    email: "user@taqtile.com",
+    password: "1234qwer",
+    birthDate: "01-01-2000",
+  };
+
+  before(async () => {
+    agent = request(`${process.env.BASE_URL}`);
+    repository = getRepository(User);
+  });
+
+  afterEach(() => {
+    repository.delete({});
+  });
+
+  const userQuery = `
+    query User($data: UserInput!) {
+      user(data: $data) {
+        id
+        name
+        email
+        birthDate
+      }
+    }
+  `;
+  it("should return an user", async () => {
+    await createUserEntity(user);
+    const savedUser = await getRepository(User).findOne({ email: user.email });
```

Comment:
Acho q vc poderia retornar o user do save direto na função `createUserEntity`. Evitaria fazer outra query aqui:

```suggestion
    const savedUser = await createUserEntity(user);
```